### PR TITLE
Avoid stomping on SatelliteDllsProjectOutputGroupDependsOn

### PR DESCRIPTION
**Pull request by AArnott**
### Context

The `SatelliteDllsProjectOutputGroupDependsOn` msbuild property is written unconditionally and without regard to its previous value. 

### Changes Made

In this change, the targets we require are *appended* to any prior value, thereby making extensibility easier.

### Testing

⛔None. What would you suggest?

This *should* theoretically be safe, since no one would reasonably set this value to anything above this point in msbuild evaluation since it would be clobbered. But if they did anyway, then it no longer being clobbered could theoretically break something. Although it seems more likely that it would *fix* or guarantee target build order where it was broken before.

### CODE REVIEW

**Review by Forgind — APPROVED** [review]
I don't think it needs a test, and it seems pretty safe from a breaking change perspective. I don't see any reason to expect this would make it more likely users would hurt themselves either, so 👍

**Comment by AArnott** [comment]
Thanks for reviewing. I don't have write permission to the repo. Can you merge, @Forgind?

**Comment by Forgind** [comment]
> Thanks for reviewing. I don't have write permission to the repo. Can you merge, @Forgind?

I normally wait for at least two approvals from maintainers. Ping in the MSBuild channel?

**Review by rainersigwald — APPROVED** [review]

**Comment by martincostello** [comment]
Might this be related to https://github.com/dotnet/sdk/issues/27890?

**Pull request merged**
